### PR TITLE
Tests for Serializer

### PR DESCRIPTION
**Pull request by arkivanov**

### CODE REVIEW

File: reaktive/src/commonTest/kotlin/com/badoo/reaktive/utils/serializer/SerializerTest.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+        serializer.accept(3)
+        serializer.accept(2)
+        serializer.accept(1)
+
+        assertValues(3, 2, 1)
+    }
+
+    @Test
+    fun value_enqueued_WHEN_sent_from_callback() {
```
Comment (CherryPerry) [inline reaktive/src/commonTest/kotlin/com/badoo/reaktive/utils/serializer/SerializerTest.kt]: What do we check with this test? That `onValue` callback was invoked and `success` was set true? We verify callback invocation in previous tests through `assertValues`.
You probably can't test directly, that `2` was added to queue. You can only do it by verifying emitted values like it's done in the next test.

Comment (arkivanov) [reply reaktive/src/commonTest/kotlin/com/badoo/reaktive/utils/serializer/SerializerTest.kt]: I want to verify that when you are submitting a new value while processing another, it doesn't call onValue recursively.

Comment (arkivanov) [reply reaktive/src/commonTest/kotlin/com/badoo/reaktive/utils/serializer/SerializerTest.kt]: And there is a separate test to verify that such a value is processed eventually.

Comment (arkivanov) [reply reaktive/src/commonTest/kotlin/com/badoo/reaktive/utils/serializer/SerializerTest.kt]: I will rename the test

Comment (arkivanov) [reply reaktive/src/commonTest/kotlin/com/badoo/reaktive/utils/serializer/SerializerTest.kt]: @CherryPerry renamed

**Pull request merged**